tests/array/array.c

```diff
@@ -34,6 +34,8 @@ int main(const int argc, const char *argv[]) {
   b = a->clone(a);
   b->copy_out(b, rb);
   AERO_ASSERT(rb[2] == 94.32);
+  AERO_ASSERT(b->data(b)[2] == 94.32);
```

Comment:
We should be careful when using floating point comparisons in assertions, particularly because we claim to offer selectable precisions. It's safer to use something like `AERO_ASSERT(fabs(b->data(b)[2] - 94.32) < tol)` here, where `tol` is a tolerance in the neighborhood of `1e-15` for double precision accuracy and `1e-8` for single precision.

Reply:
should we add an `AERO_ASSERT_ALMOST_EQUAL()` macro?

Reply:
That might be good. Or perhaps a helper function `nearly_equal(x, y, tol)` that can be used as a predicate.

Reply:
added this as issue #23 as it affects existing tests as well